Mathlib/CategoryTheory/Localization/SmallHom.lean

```diff
@@ -0,0 +1,202 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Localization.Predicate
+import Mathlib.Logic.Small.Defs
+
+/-!
+# Shrinking morphisms in localized categories
+
+Given a class of morphisms `W : MorphismProperty C`, and two objects `X` and `Y`,
+we introduce a type-class `HasSmallLocalizedHom.{w} W X Y` which expresses
+that in the localized category with respect to `W`, the type of morphisms from `X`
+to `Y` is `w`-small for a certain universe `w`. Under this assumption,
+we define `SmallHom.{w} W X Y : Type w` as the shrunk type. For any localization
+functor `L : C ⥤ D` for `W`, we provide a bijection
+`SmallHom.equiv.{w} W L : SmallHom.{w} W X Y ≃ (L.obj X ⟶ L.obj Y)` that is compatible
+with the composition of morphisms.
+
+-/
+
+universe w w' w'' v u u' u''
+
+namespace CategoryTheory
+
+variable {C : Type u} [Category.{v} C] (W : MorphismProperty C)
+  {D : Type u'} [Category.{w'} D] (L : C ⥤ D) [L.IsLocalization W]
+  {D' : Type u''} [Category.{w''} D'] (L' : C ⥤ D') [L'.IsLocalization W] (X Y Z : C)
+
+namespace MorphismProperty
+
+/-- This property holds if the type of morphisms between `X` and `Y`
+in the localized category with respect to `W : MorphismProperty C`
+is small. -/
+class HasSmallLocalizedHom : Prop where
+  small : Small.{w} (W.Q.obj X ⟶ W.Q.obj Y)
+
+variable {X Y Z}
+
+/-- Bijection between types of morphisms in two localized categories
+for the same class of morphisms `W`. -/
+noncomputable def localizationsHomEquiv :
```

Comment:
I don't think this belongs here?

Reply:
Indeed, I made generalized this construction `Localization.homEquiv` in #13956.